zebedee-cms/src/test/java/com/github/onsdigital/zebedee/model/publishing/NotificationPayloadCacheApiTest.java

```diff
@@ -0,0 +1,86 @@
+                // case 6(a): remove /data
+                {"/economy/grossdomesticproductgdp/timeseries/abmi/pn2/data", "/economy/grossdomesticproductgdp/timeseries/abmi/pn2"},
+                {"/economy/grossdomesticproductgdp/something/data", "/economy/grossdomesticproductgdp/something"},
+                {"/economy/grossdomesticproductgdp/something/data/", "/economy/grossdomesticproductgdp/something"},
+                {"/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/data", "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes"},
+                {"/economy/inflationandpriceindices/bulletins/producerpriceinflations/latest/data", "/economy/inflationandpriceindices/bulletins/producerpriceinflations/latest"},
+                // case 6(b): remove file type & file extension
+                {"/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023/upload-pricequotes202309.csv", "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023"},
+                // case 7(a): timeseries => keep up to 2 after type
+                {"/economy/grossdomesticproductgdp/timeseries/ihyq", "/economy/grossdomesticproductgdp/timeseries/ihyq"},
+                {"/economy/grossdomesticproductgdp/timeseries/abmi/pn2", "/economy/grossdomesticproductgdp/timeseries/abmi/pn2"},
+                // case 7b): timeseries => remove linechartconfig
+                {"/economy/grossdomesticproductgdp/timeseries/abmi/pn2/linechartconfig", "/economy/grossdomesticproductgdp/timeseries/abmi/pn2"},
+                {"/economy/grossdomesticproductgdp/timeseries/linechartconfig", "/economy/grossdomesticproductgdp/timeseries"},
+                // case 8: datasets => keep up to 2 after type
+                {"/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes", "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes"},
+                {"/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023", "/economy/inflationandpriceindices/datasets/consumerpriceindicescpiandretailpricesindexrpiitemindicesandpricequotes/pricequotesseptember2023"},
+                // Other tests
+                {"/economy/grossdomesticproductgdp", "/economy/grossdomesticproductgdp"},
+                {"/file?uri=/economy%5B/test%2Ftest/economy", "/economy[/test/test/economy"},
+                {"http://localhost:8080/economy/inflation/anything.xls", "/economy/inflation"},
```

Comment:
Are these an actual use case? Do you ever get full URLs? I thought that it was only paths and query strings. So no protocol or domain, or port.

Reply:
no, or I was not sure at this point. Maybe better to remove this this and also remove the method that removes the protocol and host from an url

Reply:
have removed the unneeded code and removed the tests